types/index.d.ts

```diff
@@ -2,1415 +2,1561 @@ declare var RA: RamdaAdjunct.Static;
+     */
+    trimCharsStart(chars: string, value: string): string;
+    trimCharsStart(chars: string): (value: string) => string;
+
+    /**
+     * The function pads the current string with a given string
+     * (repeated, if needed) so that the resulting string reaches a given length.
+     * The padding is applied from the end of the current string.
+     */
+    padCharsEnd(padString: string, targetLength: number, value: string): string;
+    padCharsEnd(
+      padString: string,
+      targetLength: number
+    ): (value: string) => string;
+    padCharsEnd(
+      padString: string
+    ): (targetLength: number, value: string) => string;
+
+    /**
+     * The function pads the current string with an empty string
+     * so that the resulting string reaches a given length.
+     * The padding is applied from the end of the current string.
+     */
+    padEnd(targetLength: number, value: string): string;
+    padEnd(targetLength: number): (value: string) => string;
+
+    /**
+     * The function pads the current string with a given string
+     * (repeated, if needed) so that the resulting string reaches a given lenght.
+     * The padding is applied to the start of the current string.
+     */
+    padCharsStart(
+      padString: string,
+      targetLength: number,
+      value: string
+    ): string;
+    padCharsStart(
+      padString: string,
+      targetLength: number
+    ): (value: string) => string;
+    padCharsStart(
+      padString: string
+    ): (targetLength: number, value: string) => string;
+
+    /**
+     * Pads string on the left side if it's shorter than length.
+     */
+    padStart(targetLength: number, value: string): string;
+    padStart(targetLength: number): (value: string) => string;
+
+    /**
+     * Sort a list of objects by a list of props (if first prop value is equivalent, sort by second, etc).
+     */
+    sortByProps(props: string[], list: object[]): object[];
+    sortByProps(props: string[]): (list: object[]) => object[];
+
+    /**
+     * When given a number n and an array, returns an array containing every nth element.
+     */
+    skipTake<T>(n: number, list: T[]): T[];
+    skipTake<T>(n: number): (list: T[]) => T[];
+
+    /**
+     * Determine if input value is an indexed data type.
+     */
+    isIndexed(val: any): val is string | any[];
+
+    /**
+     * Invokes the method at path of object with given arguments.
+     */
+    invokeArgs(
+      pathToMethod: string[],
+      args: Array<string | number>,
+      obj: object
+    ): any;
+    invokeArgs(
+      pathToMethod: string[],
+      args: Array<string | number>
+    ): (obj: object) => any;
+    invokeArgs(
+      pathToMethod: string[]
+    ): (args: Array<string | number>, obj: object) => any;
+
+    /**
+     * Converts double-precision 64-bit binary format IEEE 754 to signed 32 bit integer number.
+     */
+    toInteger32(n: number): number;
+    toInt32(n: number): number; // alias
+
+    /**
```

Comment:
This whole file has been changed. Probably some formatting happened. This file cannot be reviewed.

Reply:
Will commit a cleaned up version. You might want to also use prettier for the TypeScript files, though - I just saw that the repo is using prettier and only noticed now that not only is prettier not used for TypeScript, but there is even a difference in indentation settings.

Reply:
Agree. In past the `eslint` did not support TypeScript. Now it does and we're fine just using `eslint` + `prettier` to get the job done. I'll create a follow up issue for this.